### PR TITLE
Kill on sudden temperature jumps

### PR DESCRIPTION
**Pull request by AnHardt**
This is the version on the raw level.
Description and configuration in 'Configuration_adv.h'

### CODE REVIEW

**Comment by clefranc** [comment]
I'm testing this right now...

File: Marlin/temperature.cpp [hunk 1]
```diff
@@ -782,6 +786,38 @@ static float analog2tempBed(int raw) {
...
+//      SERIAL_PROTOCOLPGM("jumptest: "); SERIAL_PROTOCOL(last_temperature_raw[e+1]); SERIAL_PROTOCOLPGM(":"); SERIAL_PROTOCOLLN(((e < 0) ? current_temperature_bed_raw : current_temperature_raw[e]));
+      last_temperature_raw[e+1] = ((e < 0) ? current_temperature_bed_raw : current_temperature_raw[e]);
+      init_jump[e+1] = false;
+      return;
+    }
+    else {
+//      SERIAL_PROTOCOLPGM("jumptest: "); SERIAL_PROTOCOL(last_temperature_raw[e+1]); SERIAL_PROTOCOLPGM(":"); SERIAL_PROTOCOLLN(((e < 0) ? current_temperature_bed_raw : current_temperature_raw[e]));
+
+// next is analogue to ( TRRunaway: ). kill() with adjustable message would be better
+      SERIAL_ERROR_START;
```
Comment (thinkyhead) [inline Marlin/temperature.cpp]: Here I think you can use this:

```
_temp_error(e, PSTR(MSG_THERMAL_JUMP_STOP), PSTR(MSG_ERR_THERMAL_JUMP));
```

**Comment by clefranc** [comment]
See https://github.com/MarlinFirmware/Marlin/issues/2066 for test results.

**Comment by AnHardt** [comment]
@clefranc 
I interpret your comment as you are lucky with this.
I only want to make some cosmetic corrections.
The remaining problems with the 25°C sensor can't be trapped this way because there is no jump.
A few of the other problems will be nailed down, when _temp_error() will make more a kill() than a stop().

**Comment by thinkyhead** [comment]
I wonder if, when a thermistor table starts at 25, it would be possible to treat 25 as being unknown.

**Comment by nophead** [comment]
The fundamental problem with this thermistor is its resistance is so high at room temp the ADC resolution is not enough to be able to tell if it is disconnected or not, so MINTEMP or looking for jumps can never work. Nothing will happen until it gets hot enough to give a different ADC value from 1023. The only way to detect it is not connected is to fail when it doesn't rise when heated but there will be a dead band from room temp to some arbitrary temperature where it always reads 25.

**Comment by thinkyhead** [comment]
So… it needs to wait for the heater to emerge from that "dead band" at the low end (of temperature = high end of resistance) before beginning testing for minimum-rise-over-time. If it takes too long to come out of the dead region, only then assume a bad/dead/detached thermistor.

It sounds like we also have to do some special handling for cases when the hot end heater is off but it's rising due to ambient heat from the bed. In a case like that, it could ignore the heating if the rise-rate is below a certain threshold. Or, only begin to worry about heating-when-the-heater-is-off when the temperature gets into the PLA melting range.

**Comment by AnHardt** [comment]
@clefranc 
Save your time and don't test this now. You will not see any improvement but drawback.
I want to rework _temp_error() first. (So many postings today :-(  )

**Comment by AnHardt** [comment]
Replaced by #2164

**Pull request closed without merging**